src/screens/Addresses/Contact/EditContactScreen.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { getHumanReadableError } from '@alephium/sdk'
+import { StackScreenProps } from '@react-navigation/stack'
+import { useEffect, useState } from 'react'
+import { Alert } from 'react-native'
+import Toast from 'react-native-root-toast'
+
+import Button from '~/components/buttons/Button'
+import SpinnerModal from '~/components/SpinnerModal'
+import { useAppDispatch, useAppSelector } from '~/hooks/redux'
+import RootStackParamList from '~/navigation/rootStackRoutes'
+import { deleteContact, persistContact } from '~/persistent-storage/wallets'
+import ContactForm from '~/screens/Addresses/Contact/ContactForm'
+import {
+  contactDeletedFromPersistentStorage,
+  contactStoredInPersistentStorage
+} from '~/store/addresses/addressesActions'
+import { selectContactById } from '~/store/addresses/addressesSelectors'
+import { Contact, ContactFormData } from '~/types/contacts'
+
+type ScreenProps = StackScreenProps<RootStackParamList, 'EditContactScreen'>
+
+const EditContactScreen = ({ navigation, route: { params } }: ScreenProps) => {
+  const contact = useAppSelector((s) => selectContactById(s, params.contactId))
+  const dispatch = useAppDispatch()
+
+  const [loading, setLoading] = useState(false)
+  const [isDeleting, setIsDeleting] = useState(false)
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerRight: () => (
+        <Button
+          title="Delete"
+          type="transparent"
+          variant="alert"
+          onPress={() => {
+            Alert.alert('Deleting contact', 'Are you sure you want to delete this contact?', [
+              { text: 'Cancel' },
+              {
+                text: 'Delete',
+                onPress: async () => {
+                  setIsDeleting(true)
+
+                  try {
+                    await deleteContact(params.contactId)
```

Comment:
This function call should always be followed by the dispatch of `contactDeletedFromPersistentStorage` right? In this case, why not dispatch directly from within the `deleteContact` function?